gcloud-java-bigquery/src/test/java/com/google/gcloud/bigquery/JobInfoTest.java

```diff
@@ -0,0 +1,370 @@
+    assertNotNull(COPY_JOB.toPb().getConfiguration().getCopy());
+    assertNull(COPY_JOB.toPb().getConfiguration().getExtract());
+    assertNull(COPY_JOB.toPb().getConfiguration().getLoad());
+    assertNull(COPY_JOB.toPb().getConfiguration().getQuery());
+    assertEquals(COPY_JOB_STATISTICS, JobStatistics.fromPb(COPY_JOB.statistics().toPb()));
+    compareJobInfo(COPY_JOB, JobInfo.fromPb(COPY_JOB.toPb()));
+    assertTrue(JobInfo.fromPb(COPY_JOB.toPb()).configuration() instanceof CopyJobConfiguration);
+    assertNull(EXTRACT_JOB.toPb().getConfiguration().getCopy());
+    assertNotNull(EXTRACT_JOB.toPb().getConfiguration().getExtract());
+    assertNull(EXTRACT_JOB.toPb().getConfiguration().getLoad());
+    assertNull(EXTRACT_JOB.toPb().getConfiguration().getQuery());
+    assertEquals(EXTRACT_JOB_STATISTICS, JobStatistics.fromPb(EXTRACT_JOB.statistics().toPb()));
+    compareJobInfo(EXTRACT_JOB, JobInfo.fromPb(EXTRACT_JOB.toPb()));
+    assertTrue(
+        JobInfo.fromPb(EXTRACT_JOB.toPb()).configuration() instanceof ExtractJobConfiguration);
+    assertTrue(JobInfo.fromPb(EXTRACT_JOB.toPb()).statistics() instanceof ExtractStatistics);
+    assertNull(LOAD_JOB.toPb().getConfiguration().getCopy());
+    assertNull(LOAD_JOB.toPb().getConfiguration().getExtract());
+    assertNotNull(LOAD_JOB.toPb().getConfiguration().getLoad());
+    assertNull(LOAD_JOB.toPb().getConfiguration().getQuery());
+    assertEquals(LOAD_JOB_STATISTICS, JobStatistics.fromPb(LOAD_JOB.statistics().toPb()));
+    compareJobInfo(LOAD_JOB, JobInfo.fromPb(LOAD_JOB.toPb()));
+    assertTrue(JobInfo.fromPb(LOAD_JOB.toPb()).configuration() instanceof LoadJobConfiguration);
+    assertTrue(JobInfo.fromPb(LOAD_JOB.toPb()).statistics() instanceof LoadStatistics);
+    assertNull(QUERY_JOB.toPb().getConfiguration().getCopy());
+    assertNull(QUERY_JOB.toPb().getConfiguration().getExtract());
+    assertNull(QUERY_JOB.toPb().getConfiguration().getLoad());
+    assertNotNull(QUERY_JOB.toPb().getConfiguration().getQuery());
+    assertEquals(QUERY_JOB_STATISTICS, JobStatistics.fromPb(QUERY_JOB.statistics().toPb()));
+    compareJobInfo(QUERY_JOB, JobInfo.fromPb(QUERY_JOB.toPb()));
+    assertTrue(JobInfo.fromPb(QUERY_JOB.toPb()).configuration() instanceof QueryJobConfiguration);
+    assertTrue(JobInfo.fromPb(QUERY_JOB.toPb()).statistics() instanceof QueryStatistics);
+  }
+
+  @Test
+  public void testSetProjectId() {
+    CopyJobConfiguration copyConfiguration = COPY_JOB.setProjectId("p").configuration();
+    assertEquals("p", copyConfiguration.destinationTable().project());
+    for (TableId sourceTable : copyConfiguration.sourceTables()) {
+      assertEquals("p", sourceTable.project());
+    }
+    ExtractJobConfiguration extractConfiguration = EXTRACT_JOB.setProjectId("p").configuration();
+    assertEquals("p", extractConfiguration.sourceTable().project());
+    LoadConfiguration loadConfiguration = LOAD_JOB.setProjectId("p").configuration();
```

Comment:
I see no reason for this error but I made a change in  73250ea that should fix your problem. Could you double check?